crates/autopilot/src/run_loop.rs

```diff
@@ -348,6 +365,32 @@ impl RunLoop {
         .collect()
     }
 
+    /// Settlement is allowed to contain orders that are part of the Auction or
+    /// JIT orders.
+    async fn validate_settlement(
+        &self,
+        auction: &domain::Auction,
+        settlement: &Settlement,
+    ) -> bool {
+        let auction_orders = auction.orders.iter().map(|o| o.uid).collect::<HashSet<_>>();
+
+        // all JIT orders will be non_auction_orders
+        let non_auction_orders = settlement
+            .order_uids()
+            .filter(|order| !auction_orders.contains(order))
+            .cloned()
+            .collect::<Vec<_>>();
+
+        // make sure all non-auction orders are also non-database
+        self.persistence
+            .orders_do_not_exist(&non_auction_orders)
+            .await
+            .unwrap_or_else(|err| {
+                tracing::warn!(?err, "failed to check if all orders exist");
```

Comment:
These checks are pretty important so they should fail loudly IMO (e.g. `tracing::error!`)
```suggestion
                tracing::warn!(?err, "failed to check if all unexpected orders are JIT orders");
```